docsite/docs/testing.md

```diff
@@ -0,0 +1,173 @@
+---
+id: testing_commands
+title: Testing Commands
+sidebar_label: Testing Commands
+---
+
+Testing is an essential step in the software development cycle. `Command` uses `ValueNotifier`s internally and this makes testing commands straight forward. Traditional approaches such as `StreamMatcher` for asynchronous testing is not applicable. A `ValueNotifier` is essentially a `ChangeNotifier` whose changes can be listened to and verified. The approach we used internally is store these results in a `List` and then compare this against an expected `List` of values. Generally we will have a `Command` which is already functional and we write test against it to verify its functional aspects. In  case of widget testing it would be preferable to mock a command and test the corresponding UI elements independently. Lets see these two cases in the upcoming sections.
+
+## Testing Commands
+
+### Command to be tested
+
+Let us assume you want to test an asynchronous command which accepts a name and returns a greeting message after 2 seconds. Following is trivial implementation of such a command.
+
+```dart
+Command greetingCommand =
+    Command.createAsync<String, String>((String name) async {
+  await Future.delayed(Duration(seconds: 2));
+  return 'Hello $name! Welcome.';
+}, 'Hello! Welcome.');
+```
+
+### Test changes in `value`
+
+As a simple scenario lets test what is exposed by the `Command` which by itself is a `ValueNotifier` and can be listened to changes in its `value`.
+
+```dart
+test('Greeting Command returns correct messages', () async {
+    // A list that holds the greetings returned by the command.
+    List<String> greetings = [];
+
+    greetingCommand.addListener(() {
+      greetings.add(greetingCommand.value);
+    });
+
+    // Expect the initial Value is correct
+    expect(greetingCommand.value, 'Hello! Welcome.');
+
+    // Execute the command with the name Foo.
+    greetingCommand.execute('Alice');
+
+    // Wait for the command to execute.
+    await Future.delayed(Duration(seconds: 2));
```

Comment:
actually you don't need the await IMHO because the command is executed synchronously

Reply:
Since the next `expect` statement is synchronous and compares the `value`, it fails if I don't `await`. I tested it as part of our main tests. I didn't just commit it as it would be look out of place.